### ISSUE TITLE
Add menubar role for Platform Header dropdown

### ISSUE THREAD

**Issue by joshblack** [carbon-design-system/carbon#1270]
Looking at the WAI ARIA practices site, there is a reference to a menu or menu bar that we will need to add to our platform header: https://www.w3.org/TR/wai-aria-practices/#menu. There is also an example implementation available here: https://www.w3.org/TR/wai-aria-practices/examples/menubar/menubar-1/menubar-1.html

This issue is for updating the existing markup with the correct a11y-related roles or attributes, adjusting any styles as needed.
